processmanager/manager.go

```diff
@@ -300,6 +295,33 @@ func (pm *ProcessManager) ConvertTrace(trace *host.Trace) (newTrace *libpf.Trace
 	return newTrace
 }
 
+// findMappingForTrace locates the mapping for a given host trace.
+func (pm *ProcessManager) findMappingForTrace(pid util.PID, fid host.FileID,
+	addr libpf.AddressOrLineno) (m Mapping, found bool) {
```

Comment:
Is there a reason for not returning a `*Mapping` here?

Reply:
Yes: `findMappingForTrace` takes the `pm.mu` lock and I don't want pointers to the original mapping from the map that are protected by that lock to escape from this function after we released it. I think it would probably be fine in current impl (mappings are never mutated after creation, AFAIK), but it seems like an avoidable risk if we ever rework how mappings are populated.

Reply:
OK, my follow-up question is: why store the mappings as pointers in the first place?

Reply:
With this PR the mappings are now kept in two separate maps with different index keys (`mappings` and `mappingsByFileID`). Keeping them as pointers allows using the same instance in both map, reducing memory usage a bit.